File: c-010/Main.java
import java.util.ArrayList;
import java.util.List;

public class Main {

    public static boolean theCoolerIsPrime(int number, List<Integer> primelist) {
        int limitcheck = number;
        for (int i : primelist) {
            if (i > limitcheck) {
                return true;
            }
            if (number % i == 0) {
                return false;
            } else {
                limitcheck = number / i;
            }
        }
        return true;
    }
    
    public static long primeSumUpto(int limit) {
        long result = 0;
        List<Integer> primes = new ArrayList<>();
        for (int i = 2; i < limit; i++) {
            if (theCoolerIsPrime(i, primes)) {
                primes.add(i);
                result += i;
            }
        }
        return result;
    }    

    public static void main(String[] args) {
        System.out.println(primeSumUpto(2_000_000));
    }
}

File: c-010/main.rs
fn the_cooler_is_prime(number: i32, primelist: &Vec<i32>) -> bool {
	let mut limitcheck = number;
	for i in primelist {
        if *i > limitcheck {
			return true;
		}
		if number % i == 0 {
			return false;
		} else {
			limitcheck = number / i;
		}
	}
	return true
}

fn prime_sum_upto(limit: i32) -> i64 {
	let mut primes = Vec::new();
	let mut total: i64 = 0;
	for i in 2..limit {
		if the_cooler_is_prime(i, &primes) {
			primes.push(i);
			total += i as i64;
		}
	}
	return total;
}

fn main() {
	println!("{}", prime_sum_upto(2_000_000));
}
